pages/linux/hashcat.md

```diff
@@ -0,0 +1,32 @@
+# hashcat
+
+> Fast and advanced password recovery tool.
+> More information: <https://hashcat.net/hashcat/>.
+
+- Brute-force attack with default hashcat mask:
+
+`hashcat --hash-type {{id_hash_type}} --attack-mode 3 {{hash}}`
+
+- Brute-force attack with a known pattern of 4 digits:
+
+`hashcat --hash-type {{id_hash_type}} --attack-mode 3 {{hash}} ?d?d?d?d`
+
+- Brute-force attack with max length of 8 chars using the complete charset (numbers + letters + special chars):
+
+`hashcat --hash-type {{id_hash_type}} --attack-mode 3 --increment {{hash}} ?a?a?a?a?a?a?a?a`
```

Comment:
What does `?a?a?a?a?a?a?a?a` mean?

Reply:
`?a` is a placeholder for any ASCII symbol (<https://manned.org/hashcat#head7>). `?a?a?a?a?a?a?a?a` means there are eight of them.

Reply:
However it's still kind of a variable so we should use:
```suggestion
`hashcat --hash-type {{id_hash_type}} --attack-mode 3 --increment {{hash}} {{?a?a?a?a?a?a?a?a}}`
```

Reply:
I think we should also put it in quotes, because most shells interpret `?` as a special character.  For me, `echo ?` doesn't work.  Only `echo "?"`.

Reply:
Fixed with the brackets but without the quotes because I do not think they are needed in this case. @bl-ue , do you run into any issue running the hashcat command in your shell?

Reply:
It works without quotes in `bash`, but not in `zsh` or `fish`.

Reply:
ok, then fixed.